fairseq/tasks/hubert_pretraining.py

```diff
@@ -107,19 +107,19 @@ class HubertPretrainingTask(FairseqTask):
     def __init__(
         self,
         cfg: HubertPretrainingConfig,
-        dictionaries: Dict[str, Dictionary],
     ) -> None:
         super().__init__(cfg)
 
         logger.info(f"current directory is {os.getcwd()}")
         logger.info(f"HubertPretrainingTask Config {cfg}")
-        self._dictionaries = dictionaries
+        
+        self.state.add_factory("dictionaries", lambda: self.dictionaries_factory(cfg))
+        self.state.add_factory("target_dictionary", self.target_dictionary_factory)
```

Comment:
Since target dictionary can be retrieved from dictionaries, this is not required.